pkg/cmd/repo/create/create.go

```diff
@@ -349,90 +333,324 @@ func createRun(opts *CreateOptions) error {
-				if !createLocalDirectory && (gitIgnoreTemplate == "" && repoLicenseTemplate == "") {
-					err := prompt.Confirm(fmt.Sprintf(`Create a local project directory for "%s"?`, ghrepo.FullName(repo)), &createLocalDirectory)
-					if err != nil {
-						return err
-					}
-				} else if !createLocalDirectory && (gitIgnoreTemplate != "" || repoLicenseTemplate != "") {
-					err := prompt.Confirm(fmt.Sprintf(`Clone the remote project directory "%s"?`, ghrepo.FullName(repo)), &createLocalDirectory)
-					if err != nil {
-						return err
-					}
-				}
+		if opts.LicenseTemplate == "" && opts.GitIgnoreTemplate == "" {
+			// cloning empty repository or template
+			checkoutBranch := ""
+			if opts.Template != "" {
+				checkoutBranch = templateRepoMainBranch
 			}
-			if createLocalDirectory && (gitIgnoreTemplate == "" && repoLicenseTemplate == "") {
-				path := repo.RepoName()
-				checkoutBranch := ""
-				if opts.Template != "" {
-					// NOTE: we cannot read `defaultBranchRef` from the newly created repository as it will
-					// be null at this time. Instead, we assume that the main branch name of the new
-					// repository will be the same as that of the template repository.
-					checkoutBranch = templateRepoMainBranch
-				}
-				if err := localInit(opts.IO, remoteURL, path, checkoutBranch); err != nil {
-					return err
-				}
-				if isTTY {
-					fmt.Fprintf(stderr, "%s Initialized repository in \"%s\"\n", cs.SuccessIcon(), path)
-				}
-			} else if createLocalDirectory && (gitIgnoreTemplate != "" || repoLicenseTemplate != "") {
-				_, err := git.RunClone(remoteURL, []string{})
-				if err != nil {
-					return err
-				}
+			if err := localInit(opts.IO, remoteURL, repo.RepoName(), checkoutBranch); err != nil {
+				return err
 			}
+		} else if _, err := git.RunClone(remoteURL, []string{}); err != nil {
+			return err
 		}
 
-		return nil
+		if isTTY {
+			fmt.Fprintf(opts.IO.Out,
+				"%s Initialized repository in \"%s\"\n",
+				cs.SuccessIcon(), repo.RepoName())
+		}
 	}
-	fmt.Fprintln(opts.IO.Out, "Discarding...")
+
 	return nil
 }
 
-func interactiveGitIgnore(client *http.Client, hostname string) (string, error) {
+// create repo on remote host from existing local repo
+func createFromLocal(opts *CreateOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	cs := opts.IO.ColorScheme()
+	isTTY := opts.IO.IsStdoutTTY()
+	stdout := opts.IO.Out
+
+	cfg, err := opts.Config()
+	if err != nil {
+		return err
+	}
+	host, err := cfg.DefaultHost()
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		opts.Source, err = interactiveSource()
+		if err != nil {
+			return err
+		}
+	}
+
+	repoPath := opts.Source
+
+	var baseRemote string
+	if opts.Remote == "" {
+		baseRemote = "origin"
+	} else {
+		baseRemote = opts.Remote
+	}
+
+	projectDir, projectDirErr := git.GetDirFromPath(repoPath)
+	if projectDirErr != nil {
+		return projectDirErr
+	}
+	if projectDir != ".git" {
```

Comment:
I don't think we should error out if `projectDir` isn't ".git". The value doesn't matter at all. The command succeeded: that means `repoPath` is a git directory.

Reply:
iirc the command `git -C $source rev-parse --git-dir` still succeeds in a git subdirectory, and we only want to proceed if the path is the top level git directory. This was our "hack" to check that. Is there a better way @mislav ?

Reply:
And as far as the command goes, it creates a git repository on github and then it tries to add a "remote" and it fails because it's not a top level git directory

Reply:
Ah, I see! I did not understand that goal from looking at the code. Perhaps `git rev-parse --show-cdup` could be used to check whether you are in a subdirectory of a local git directory? But if `--show-cdup` was added to git only relatively recently (I do not know how old it is) then your method is fine as well.